src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/SimpleTypeModelBinder.cs

```diff
@@ -12,6 +13,18 @@ namespace Microsoft.AspNetCore.Mvc.ModelBinding.Binders
     /// </summary>
     public class SimpleTypeModelBinder : IModelBinder
     {
+        private readonly TypeConverter _typeConverter;
+
+        public SimpleTypeModelBinder(Type type)
+        {
+            if (type == null)
+            {
+                throw new ArgumentNullException(nameof(type));
+            }
+
+            _typeConverter = TypeDescriptor.GetConverter(type);
```

Comment:
You need to check `CanConvert` here and fall back to the string type converter. Take a closer look at the code you're replacing.